admin/class-create-block-theme-admin.php

```diff
@@ -0,0 +1,428 @@
+ * The admin-specific functionality of the plugin.
+ *
+ * @package    Create_Block_Theme 
+ * @subpackage Create_Block_Theme/admin
+ * @author     Automattic
+ */
+class Create_Block_Theme_Admin {
+
+	/**
+	 * Initialize the class and set its properties.
+	 */
+	public function __construct() {
+		add_action( 'admin_menu', [ $this, 'create_admin_menu' ] );
+		add_action( 'admin_init', [ $this, 'blockbase_save_theme' ] );
+	}
+
+	function create_admin_menu() {
+		$page_title=__('Create Block Theme', 'create-block-theme');
+		$menu_title=__('Create Block Theme', 'create-block-theme');
+		add_theme_page( $page_title, $menu_title, 'edit_theme_options', 'create-block-theme', [ $this, 'create_admin_form_page' ] );
+	}
+
+	/**
+	 * Export activated child theme
+	 */
+	function export_child_theme( $theme ) {
+		$theme['slug'] = wp_get_theme()->get( 'TextDomain' );
+
+		// Create ZIP file in the temporary directory.
+		$filename = tempnam( get_temp_dir(), $theme['slug'] );
+		$zip = $this->create_zip( $filename );
+
+		$zip = $this->copy_theme_to_zip( $zip );
+		$zip = $this->add_templates_to_zip( $zip, 'current' );
+		$zip = $this->add_theme_json_to_zip( $zip, 'current' );
+
+		$zip->close();
+
+		header( 'Content-Type: application/zip' );
+		header( 'Content-Disposition: attachment; filename=' . $theme['slug'] . '.zip' );
+		header( 'Content-Length: ' . filesize( $filename ) );
+		flush();
+		echo readfile( $filename );
+		die();
+	}
+
+	/**
+	 * Create a sibling theme of the activated theme
+	 */
+	function create_sibling_theme( $theme ) {
+		// Sanitize inputs.
+		$theme['name'] = sanitize_text_field( $theme['name'] );
+		$theme['description'] = sanitize_text_field( $theme['description'] );
+		$theme['uri'] = sanitize_text_field( $theme['uri'] );
+		$theme['author'] = sanitize_text_field( $theme['author'] );
+		$theme['author_uri'] = sanitize_text_field( $theme['author_uri'] );
+		$theme['slug'] = sanitize_title( $theme['name'] );
+		$theme['template'] = wp_get_theme()->get( 'Template' );
+
+		// Create ZIP file in the temporary directory.
+		$filename = tempnam( get_temp_dir(), $theme['slug'] );
+		$zip = $this->create_zip( $filename );
+
+		$zip = $this->add_templates_to_zip( $zip, 'current' );
+		$zip = $this->add_theme_json_to_zip( $zip, 'current' );
+
+		// Add readme.txt.
+		$zip->addFromString(
+			'readme.txt',
+			$this->build_readme_txt( $theme )
+		);
+
+		// Add style.css.
+		$zip->addFromString(
+			'style.css',
+			$this->build_child_style_css( $theme )
+		);
+
+		// Add screenshot.png.
+		$zip->addFile(
+			'screenshot.png',
+			$theme['slug'] . '/screenshot.png'
+		);
+
+		$zip->close();
+
+		header( 'Content-Type: application/zip' );
+		header( 'Content-Disposition: attachment; filename=' . $theme['slug'] . '.zip' );
+		header( 'Content-Length: ' . filesize( $filename ) );
+		flush();
+		echo readfile( $filename );
+		die();
+	}
+
+	/**
+	 * Create a child theme of the activated theme
+	 */
+	function create_child_theme( $theme ) {
```

Comment:
this function is very similar to `create_sibling_theme`, maybe there's a way we can share the common parts of the code?

Reply:
create_sibling_theme isn't quite done yet, and I suspect as it resolves it's going to take on aspects of both create_child_theme and export_theme.  I think there's common logic we can extract from all four "theme creation" functions though so maybe we can "piggyback" that from one into another somehow.